client/allocrunnerv2/taskrunner/lifecycle.go

```diff
@@ -66,8 +66,8 @@ func (tr *TaskRunner) Kill(ctx context.Context, event *structs.TaskEvent) error
 		return ErrTaskNotRunning
 	}
 
-	// Emit the event
-	tr.EmitEvent(event)
+	// Append the event (it will get emitted by UpdateState)
+	tr.AppendEvent(event)
```

Comment:
I think this should stay emit event since you can have long kill timeouts and knowing it was being killed in the meantime would otherwise be lost